pgmpy/factors/ContinuousNode.py

```diff
@@ -0,0 +1,230 @@
+from scipy import integrate
+from scipy.stats import rv_continuous
+
+import numpy as np
+
+
+class ContinuousNode(rv_continuous):
+    """
+    Class for continuous node representation. This is a subclass of
+    scipy.stats.rv_continuous.
+    This allows representation of user defined continuous distribution
+    by specifying a function to compute the probability density function
+    of the distribution.
+    All methods of the scipy.stats.rv_continuous class can be used on
+    the objects.
+    This supports an extra method to discretize the continuous distribution
+    into a discrete factor using various methods.
+    """
+    def __init__(self, pdf, lb=None, ub=None):
+        """
+        Parameters
+        ----------
+        pdf : function
+        The user defined probability density function.
+
+        lb : float, optional
+        Lower bound of the support of the distribution, default is minus
+        infinity.
+
+        ub : float, optional
+        Upper bound of the support of the distribution, default is plus
+        infinity.
+
+        Examples
+        --------
+        >>> from pgmpy.factors import ContinuousNode
+        >>> custom_pdf = lambda x : 0.5 if x > -1 and x < 1 else 0
+        >>> node = ContinuousNode(custom_pdf, -3, 3)
+        """
+        self.pdf = pdf
+        super(ContinuousNode, self).__init__(momtype=0, a=lb, b=ub)
+
+    def _pdf(self, *args):
+        """
+        Defines the probability density function of the given
+        continuous variable.
+        """
+        return self.pdf(*args)
+
+    def _rounding_method(self, frm, to, step):
```

Comment:
@ankurankan I think we can have all the continuous part in separate directory "continuous" with sub-directories  "discretizers" and "represenataion"  or may be "factor" instead of represenatation.